com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,150 @@
+        CloudControllerClient controllerClient, String spaceId) {
+        Map<CloudServiceExtended, ServiceOperation> servicesOperation = new HashMap<>();
+        existingServices.forEach(service -> {
+            Map<String, Object> serviceInstanceEntity = getServiceInstanceEntity(controllerClient, service, spaceId);
+            ServiceOperation serviceOperationInProgress = getServiceOperationInProgress(serviceInstanceEntity);
+            if (serviceOperationInProgress != null) {
+                servicesOperation.put(service, serviceOperationInProgress);
+            }
+        });
+        return servicesOperation;
+    }
+
+    private Map<String, Object> getServiceInstanceEntity(CloudControllerClient client, CloudService service, String spaceId) {
+        return serviceInstanceGetter.getServiceInstanceEntity(client, service.getName(), spaceId);
+    }
+
+    private ServiceOperation getServiceOperationInProgress(Map<String, Object> serviceInstanceEntity) {
+        ServiceOperation lastOpertaion = getLastOperation(serviceInstanceEntity);
+        if (lastOpertaion != null && lastOpertaion.getState() == ServiceOperationState.IN_PROGRESS) {
+            return lastOpertaion;
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    private ServiceOperation getLastOperation(Map<String, Object> cloudServiceInstance) {
+        Map<String, Object> lastOperationAsMap = (Map<String, Object>) MapUtils.getMap(cloudServiceInstance, LAST_SERVICE_OPERATION);
+        if (lastOperationAsMap == null) {
+            return null;
+        }
+        return parseServiceOperationFromMap(lastOperationAsMap);
+    }
+
+    private ServiceOperation parseServiceOperationFromMap(Map<String, Object> serviceOperation) {
+        if (serviceOperation.get(SERVICE_OPERATION_TYPE) == null || serviceOperation.get(SERVICE_OPERATION_STATE) == null) {
+            return null;
+        }
+        ServiceOperationType type = ServiceOperationType.fromString((String) serviceOperation.get(SERVICE_OPERATION_TYPE));
+        ServiceOperationState state = ServiceOperationState.fromString((String) serviceOperation.get(SERVICE_OPERATION_STATE));
+        String description = (String) serviceOperation.get(SERVICE_OPERATION_DESCRIPTION);
+        return new ServiceOperation(type, description, state);
+    }
+
+    private Map<String, ServiceOperationType> getServicesOperationTypes(
+        Map<CloudServiceExtended, ServiceOperation> servicesInProgressState) {
+        return servicesInProgressState.entrySet()
+            .stream()
+            .collect(Collectors.toMap(serviceName -> serviceName.getKey()
+                .getName(),
+                serviceOperationType -> serviceOperationType.getValue()
+                    .getType()));
+    }
+
+    private Map<String, CloudServiceExtended> getServicesData(Map<CloudServiceExtended, ServiceOperation> servicesInProgressState) {
```

Comment:
Why is this? Dont get it - you have the CloudServiceExtended - just extract it from the map.

Reply:
fixed